packages/components/input-otp/src/use-input-otp.ts

```diff
@@ -127,6 +127,14 @@ export function useInputOtp(originalProps: UseInputOtpProps) {
   const inputRef = useDOMRef<HTMLInputElement>(ref);
   const baseDomRef = useDOMRef<HTMLDivElement>(baseRef);
 
+  const regex = new RegExp(allowedKeys);
+  let valueFromProps = props.value
+    ? props.value
+        .split("")
+        .filter((char) => regex.test(char))
+        .join("")
+    : undefined;
```

Comment:
wondering why we test each char one by one instead of just testing the whole `props.value`?